### DISCUSSION TITLE
mysqldef: json column default:  syntax error near '{}'

### DISCUSSION THREAD

**Discussion by zztkm** [sqldef/sqldef#630] [Feature requests]
## Platform

* OS: macOS
* RDBMS: MySQL
* Version: maybe v0.16.12

## --export output

```bash
CREATE TABLE `hello_json_utf8mb4` (
  `id` int NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
```

## Input SQL

```sql
CREATE TABLE `hello_json_utf8mb4` (
  `id` int NOT NULL AUTO_INCREMENT,
  `json_col` json NOT NULL DEFAULT (_utf8mb4'{}'),
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
```

## Current output

```bash
found syntax error when parsing DDL "CREATE TABLE `hello_json_utf8mb4` (
  `id` int NOT NULL AUTO_INCREMENT,
  `json_col` json NOT NULL DEFAULT (_utf8mb4'{}'),
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin": syntax error at position 121 near '{}'
```

## Expected output

```bash
Run: 'ALTER TABLE user ADD COLUMN json_col json NOT NULL ~  ;'

**Comment by zztkm** [comment]
If you know the revision policy and can tell me about it, i would be willing to submit a PR.

**Comment by k0kubun** [comment]
Sorry, I don't understand what is "the revision policy".

This is probably a bug in the parser. The lexer (`parser/token.go`) and/or the parser (`parser/parser.y`) would need to be modified. If you can open a PR, that would be pretty much appreciated.

**Comment by zztkm** [comment]
> Sorry, I don't understand what is "the revision policy".

Sorry, my English was not correct...

I wanted to ask you what I should do to fix the syntax error.

I will use the following post to work on a fix!
https://qiita.com/k0kubun/items/1b641dfd186fe46feb65

**Comment by zztkm** [comment]
memo:

mysql min_version: 8.0.13

https://dev.mysql.com/doc/refman/8.0/ja/json.html#:~:text=MySQL%208.0.13%20%E3%82%88%E3%82%8A%E5%89%8D%E3%81%AF%E3%80%81JSON%20%E3%82%AB%E3%83%A9%E3%83%A0%E3%81%AB%20NULL%20%E4%BB%A5%E5%A4%96%E3%81%AE%E3%83%87%E3%83%95%E3%82%A9%E3%83%AB%E3%83%88%E5%80%A4%E3%82%92%E5%90%AB%E3%82%81%E3%82%8B%E3%81%93%E3%81%A8%E3%81%AF%E3%81%A7%E3%81%8D%E3%81%BE%E3%81%9B%E3%82%93%E3%80%82

**Comment by gfx** [comment]
Resolved in https://github.com/sqldef/sqldef/pull/934
